packages/aws-cdk/test/api/hotswap/hotswap-test-setup.ts

```diff
@@ -60,16 +96,22 @@ export function stackSummaryOf(logicalId: string, resourceType: string, physical
 export class HotswapMockSdkProvider {
   public readonly mockSdkProvider: MockSdkProvider;
 
-  constructor() {
+  constructor(rootStackName?: string, childStacks?: string[]) {
```

Comment:
How about

```suggestion
  constructor(rootStackName?: string, childStacks: string[] = []) {
```

? Will save you the null checks on `childStacks` below.

Reply:
I think we don't actually need `childStacks` at all, I've reworked this to be better.